pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -760,13 +764,14 @@ def _handle_distribution_tree(declarative_content):
                     continue
 
                 future_relations = declarative_content.extra_data
-                update_collections = future_relations.get('collections') or {}
-                update_references = future_relations.get('references') or []
+                update_collections = future_relations.get('collections', {})
+                update_references = future_relations.get('references', [])
 
                 for update_collection, packages in update_collections.items():
-                    # need to save a collection before adding a relation
-                    update_collection.save()
-                    update_record.collections.add(update_collection)
```

Comment:
As @dkliban pointed, these lines were consuming some considerable amount of memory, so I changed to bulk creation.
good improvement in memory consumption